actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -243,10 +237,10 @@ def fragment_name_with_digest(name, virtual_path, digest_path)
 
       def fragment_for(name = {}, options = nil, &block)
         if content = read_fragment_for(name, options)
-          @view_renderer.cache_hits[@virtual_path] = :hit if defined?(@view_renderer)
+          @view_renderer.cache_hits[@current_template&.virtual_path] = :hit if defined?(@view_renderer)
```

Comment:
Before this change was `@virtual_path` nil sometimes?

Reply:
Yeah, when using inline rendering:

```
class DemoController < ActionController::Base
  def index
    render inline: "<%= @virtual_path.nil? %>"
  end
end
```